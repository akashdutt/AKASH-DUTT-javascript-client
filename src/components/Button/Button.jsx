import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styling from './style';

const propTypes = {
  color: PropTypes.string,
  disabled: PropTypes.bool,
  style: PropTypes.objectOf(PropTypes.string),
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
const defaultProps = {
  color: 'primary',
  disabled: false,
  style: {},
};
class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  render() {
    const {
      color,
      disabled,
      style,
      value,
      onClick,
      ...rest
    } = this.props;
    return (
      <>
        <button {...rest} type="button" style={{ ...styling.base, ...style }} disabled={disabled}>{value}</button>
      </>
    );
  }
}
Button.propTypes = propTypes;
Button.defaultProps = defaultProps;
export default Button;
