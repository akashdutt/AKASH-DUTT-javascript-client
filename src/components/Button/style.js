const styling = {
  base: {
    paddingTop: '10px',
    paddingBottom: '10px',
    paddingRight: '15px',
    paddingLeft: '15px',
    marginTop: '15px',
    marginRight: '10px',
  },
};
export default styling;
