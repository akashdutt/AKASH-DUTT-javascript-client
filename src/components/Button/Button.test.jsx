import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
  it('should render correctly for button mode', () => {
    const component = shallow(<Button />);
    expect(component).toMatchSnapshot();
  });
});
