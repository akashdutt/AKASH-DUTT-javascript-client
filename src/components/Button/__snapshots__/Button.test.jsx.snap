// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly for button mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    color="primary"
    disabled={false}
    style={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <button
        disabled={false}
        style={
          Object {
            "marginRight": "10px",
            "marginTop": "15px",
            "paddingBottom": "10px",
            "paddingLeft": "15px",
            "paddingRight": "15px",
            "paddingTop": "10px",
          }
        }
        type="button"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "disabled": false,
        "style": Object {
          "marginRight": "10px",
          "marginTop": "15px",
          "paddingBottom": "10px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "10px",
        },
        "type": "button",
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <button
          disabled={false}
          style={
            Object {
              "marginRight": "10px",
              "marginTop": "15px",
              "paddingBottom": "10px",
              "paddingLeft": "15px",
              "paddingRight": "15px",
              "paddingTop": "10px",
            }
          }
          type="button"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "disabled": false,
          "style": Object {
            "marginRight": "10px",
            "marginTop": "15px",
            "paddingBottom": "10px",
            "paddingLeft": "15px",
            "paddingRight": "15px",
            "paddingTop": "10px",
          },
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Button should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    color="primary"
    disabled={false}
    style={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <button
        disabled={false}
        style={
          Object {
            "marginRight": "10px",
            "marginTop": "15px",
            "paddingBottom": "10px",
            "paddingLeft": "15px",
            "paddingRight": "15px",
            "paddingTop": "10px",
          }
        }
        type="button"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "disabled": false,
        "style": Object {
          "marginRight": "10px",
          "marginTop": "15px",
          "paddingBottom": "10px",
          "paddingLeft": "15px",
          "paddingRight": "15px",
          "paddingTop": "10px",
        },
        "type": "button",
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <button
          disabled={false}
          style={
            Object {
              "marginRight": "10px",
              "marginTop": "15px",
              "paddingBottom": "10px",
              "paddingLeft": "15px",
              "paddingRight": "15px",
              "paddingTop": "10px",
            }
          }
          type="button"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "disabled": false,
          "style": Object {
            "marginRight": "10px",
            "marginTop": "15px",
            "paddingBottom": "10px",
            "paddingLeft": "15px",
            "paddingRight": "15px",
            "paddingTop": "10px",
          },
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
